### PR TITLE
replace `GetLatestFinalizedHeader` with `GetFinalizedHeader`

### PR DESCRIPTION
**Pull request by siburu**

### CODE REVIEW

File: chains/tendermint/prover.go [hunk 1]
```diff
@@ -7,11 +7,13 @@ import (
 	"time"
 
 	"github.com/cometbft/cometbft/light"
+	"github.com/cometbft/cometbft/types"
 	tmtypes "github.com/cometbft/cometbft/types"
 	"github.com/cosmos/cosmos-sdk/codec"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	clienttypes "github.com/cosmos/ibc-go/v7/modules/core/02-client/types"
 	ibcclient "github.com/cosmos/ibc-go/v7/modules/core/client"
+	"github.com/cosmos/ibc-go/v7/modules/core/exported"
```
Comment (bluele) [inline chains/tendermint/prover.go:16]: Looks like `exported` is already imported at line 17 :)

File: chains/tendermint/prover.go [hunk 2]
```diff
@@ -196,13 +193,18 @@ func (pr *Prover) UpdateLightClient() (core.Header, error) {
...
+	if height == 0 {
+		sh, err = client.Update(context.Background(), time.Now())
+	} else {
+		sh, err = client.VerifyLightBlockAtHeight(context.Background(), height, time.Now())
+	}
 	if err != nil {
 		return nil, lightError(err)
 	}
 
 	if sh == nil {
```
[the remark this answers is not in the document]
Comment (siburu) [reply chains/tendermint/prover.go]: @bluele I see. Thanks to your comment, I understand that `client.Update()` returns `(nil, nil)` if the latest block is already stored in the DB but that `client.VerifyLightBlockAtHeight` doesn't return `(nil, nil)` even at the case.

I fixed it. Please review the PR again.

File: core/provers.go [hunk 3]
```diff
@@ -48,9 +48,10 @@ type LightClient interface {
 
 // FinalityAware provides the capability to determine the finality of the chain
 type FinalityAware interface {
-	// GetLatestFinalizedHeader returns the latest finalized header on this chain
-	// The returned header is expected to be the latest one of headers that can be verified by the light client
-	GetLatestFinalizedHeader() (latestFinalizedHeader Header, err error)
+	// GetFinalizedHeader returns the finalized header on this chain corresponding to `height`.
+	// If `height` is nil, this function returns the latest finalized header.
```
Comment (bluele) [inline core/provers.go:52]: @siburu Could you fix this comment as s/nil/0/ ?

Comment (siburu) [reply core/provers.go]: Thank you! I fixed it.

**Review by bluele — APPROVED** [review]
@siburu Thanks! LGTM👍

**Pull request merged**